build.py

```diff
@@ -0,0 +1,192 @@
+    "--python", action='store_true', help='Build PyCylon wrapper')
+python_build.add_argument(
+    "--pytest", action='store_true', help='Run Python test suite')
+
+# Docker build
+java_build = parser.add_argument_group("Docker")
+java_build.add_argument("--docker", action='store_true',
+                        help='Build Cylon Docker images locally')
+
+# Paths
+parser.add_argument("-bpath", help='Build directory',
+                    default=Path(os.getcwd(), 'build'))
+parser.add_argument("-ipath", help='Install directory')
+
+args = parser.parse_args()
+
+# Variables
+
+
+def on_off(arg):
+    return "ON" if arg else "OFF"
+
+
+BUILD_CPP = args.cpp
+CPP_BUILD_MODE = "Release" if (args.release or (
+    not args.release and not args.debug)) else "Debug"
+CPP_SOURCE_DIR = str(Path(args.root, 'cpp'))
+PYTHON_SOURCE_DIR = Path(args.root, 'python', 'pycylon')
+RUN_CPP_TESTS = args.test
+RUN_PYTHON_TESTS = args.pytest
+CMAKE_FLAGS = args.cmake_flags
+CPPLINT_COMMAND = " \"-DCMAKE_CXX_CPPLINT=cpplint;--linelength=100;--headers=h,hpp;--filter=-legal/copyright,-build/c++11,-runtime/references\" " if args.style_check else ""
+
+# arrow build expects /s even on windows
+BUILD_PYTHON = args.python
+
+# docker
+BUILD_DOCKER = args.docker
+
+BUILD_DIR = str(Path(args.bpath))
+INSTALL_COMMAND = f"-DCMAKE_INSTALL_PREFIX={args.ipath}" if args.ipath else ""
+
+def print_line():
+    print("=================================================================")
+
+print_line()
+print("Build mode: " + CPP_BUILD_MODE)
+print("Build path: " + BUILD_DIR)
+print_line()
+
+# create build directory
+if not os.path.exists(BUILD_DIR):
+    print("Creating build directory...")
+    os.makedirs(BUILD_DIR)
+
+def check_status(status, task):
+    if status != 0:
+        print(f'{task} failed with a non zero exit code. Cylon build is terminating.')
+        quit()
+    else:
+        print(f'{task} completed successfully')
+
+def python_command():
+    res = subprocess.call("python3 --version")
+    if res == 0:
+        return "python3"
+
+    res = subprocess.call("python --version")
+    if res == 0:
+        return "python"
+
+    print("Python not found.")
+    quit()
+
+
+def build_cpp():
+    if not BUILD_CPP:
+        return
+
+    CONDA_PREFIX = os.getenv('CONDA_PREFIX')
+    INSTALL_COMMAND = f"-DCMAKE_INSTALL_PREFIX={args.ipath}" if args.ipath else ""
+    if not args.ipath:
+        if CONDA_PREFIX:
+            INSTALL_COMMAND = f"-DCMAKE_INSTALL_PREFIX={CONDA_PREFIX}"
+        else:
+            print("The build should be in a conda environment")
+            return
+
+    cmake_command = f'cmake -DPYCYLON_BUILD={on_off(BUILD_PYTHON)} -A x64 \
+      -DCMAKE_BUILD_TYPE={CPP_BUILD_MODE} -DCYLON_WITH_TEST={on_off(RUN_CPP_TESTS)} -DARROW_BUILD_TYPE=SYSTEM {CPPLINT_COMMAND} {INSTALL_COMMAND} \
+      {CMAKE_FLAGS} {CPP_SOURCE_DIR}'
+
+    print(cmake_command, BUILD_DIR)
+    res = subprocess.call(cmake_command, cwd=BUILD_DIR, shell=True)
```

Comment:
don't we have to forward the environment variables to the subprocess here? expecially in the conda environment?

Reply:
I think we don't need to.